sdk/client.go

```diff
@@ -108,19 +110,34 @@ func closeResponseBody(ctx context.Context, resp *http.Response) {
 
 // Takes the input http response and unmarshals the body to the input target
 func unmarshalResponseBodyExpectingStringError(response *http.Response, target interface{}) (err error) {
+	// Read the entire response body first, regardless of status code.
```

Comment:
It's unclear to me why this function was updated - it's used in many other places in this SDK so it would be good to know if this affects how those functions return information to those services consuming the function from the SDK.

Now that the changes have been implemented within the GetDataset function are these changes are still needed?